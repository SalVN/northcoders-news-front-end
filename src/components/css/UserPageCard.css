.user-page-card-box {
    margin: 20px;
    border: 1px solid lightgrey;
}

.user-page-card-background {
    background-color: #373737;
    height: 180px;
    border-top-left-radius: 5px;
    border-top-right-radius: 5px;
    text-align: left;
}

.user-card-avatar-page {
    max-height: 100px;
    max-width: 100px;
    border-radius: 50px;
    margin: 30px;
    border: 2px solid #c1002e;
    background-color: white;
}

.user-page-card-name {
    color: white;
    font-size: 28px;
    font-weight: 700;
    margin-top: 40px;
}

.user-page-card-username {
    color: lightgrey;
    font-size: 20px;
}

.avatar-container {
    text-align: center;
}

.stats-container {
    margin-top: -10px;
}

.content-container {
    background-color: lightgray;
    min-height: 75px;
}

.user-page-card-mini-title {
    margin-top: -10px;
    color: black;
    font-size: 12px;
    text-align: center;
}
.user-page-card-number {
    color: #c1002e;
    text-align: center;
    font-size: 18px;
    font-weight: 700;
}

/*.card-box {
    margin: 10px;
}

.background {
    background-color: #373737;
    height: 80px;
    border-top-left-radius: 5px;
    border-top-right-radius: 5px;
    text-align: center;
}

.user-card-avatar {
    height: 60px;
    border-radius: 50px;
    margin-top: 10px;
}

.user-card-name {
    color: #c1002e;
    font-size: 18px;
    font-weight: strong;
}

.user-card-username {
    color: grey;
    font-size: 14px;
}


}*/