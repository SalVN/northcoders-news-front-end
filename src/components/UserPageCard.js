import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './css/UserPageCard.css';

const UserPageCard = function (props) {
        return (
            <div className='user-page-card-box'>
                <div className='container-fluid'>
                    <div className='card'>
                        <div className='user-page-card-background'>
                            <div className='columns is-mobile'>
                                <div className='column is-half-mobile is-one-quarter-tablet avatar-container'>
                                    <img className='user-card-avatar-page' src={props.user.avatar_url} />
                                </div>
                                <div className='column is-half-mobile is-three-quarters-tablet'>
                                    <p className='user-page-card-name'>{props.user.name}</p>
                                    <p className='user-page-card-username'>{`@${props.user.username}`}</p>
                                </div>
                            </div>
                        </div>
                        <div className='card-content content-container'>
                            <div className='columns is-mobile'>
                                <div className='column is-2-tablet'>
                                    <div className='user-page-card-mini-title'>Popularity</div>
                                    <div className='user-page-card-number'>{props.user.comments_vote_count + props.user.articles_vote_count}</div>
                                </div>
                                <div className='column is-2-tablet'>
                                    <div className='user-page-card-mini-title'>Comments</div>
                                    <div className='user-page-card-number'>{props.user.comment_count}</div>
                                </div>
                                <div className='column is-2-tablet'>
                                    <div className='user-page-card-mini-title'>Articles</div>
                                    <div className='user-page-card-number'>{props.articlesNo}</div>
                                </div>
                                <div className='column is-2-tablet is-hidden-mobile'>
                                    <div className='user-page-card-mini-title'>Comment Votes</div>
                                    <div className='user-page-card-number'>{props.user.comments_vote_count}</div>
                                </div>
                                <div className='column is-2-tablet is-hidden-mobile'>
                                    <div className='user-page-card-mini-title'>Article Votes</div>
                                    <div className='user-page-card-number'>{props.user.articles_vote_count}</div>
                                </div>
                                <div className='column is-2-tablet is-hidden-mobile'>
                                    <div className='user-page-card-mini-title'>Ranking</div>
                                    <div className='user-page-card-number'>{props.ranking}</div>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        );
};

UserPageCard.propTypes = {
    user: PropTypes.object.isRequired,
    articlesNo: PropTypes.number.isRequired,
    ranking: PropTypes.number.isRequired
};

export default UserPageCard;