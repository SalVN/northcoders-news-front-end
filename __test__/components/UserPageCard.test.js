import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { Route, MemoryRouter } from 'react-router-dom';

import UserPageCard from '../../src/components/UserPageCard';

describe('UserPageCard', () => {
    const user = {
        _id: '59b1b18b327cce1fb043bdb1',
        username: 'northcoder',
        name: 'Awesome Northcoder',
        avatar_url: 'https://avatars3.githubusercontent.com/u/6791502?v=3&s=200',
        __v: 0
    };
    it('is a function', () => {
        expect(typeof UserPageCard).toEqual('function');
    });

    it('renders', () => {
        const enzymeWrapper = shallow(<UserPageCard
            user={user}
        />);
        expect(enzymeWrapper.children().length).toEqual(1);
    });

    it('renders correctly', () => {
        const tree = renderer.create(
            <MemoryRouter>
                <UserPageCard
                    user={user}
                />
            </MemoryRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});