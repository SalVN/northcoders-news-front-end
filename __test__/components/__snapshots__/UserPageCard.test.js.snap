// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPageCard renders correctly 1`] = `
<div
  className="user-page-card-box"
>
  <div
    className="container-fluid"
  >
    <div
      className="card"
    >
      <div
        className="user-page-card-background"
      >
        <div
          className="columns is-mobile"
        >
          <div
            className="column is-half-mobile is-one-quarter-tablet avatar-container"
          >
            <img
              className="user-card-avatar-page"
              src="https://avatars3.githubusercontent.com/u/6791502?v=3&s=200"
            />
          </div>
          <div
            className="column is-half-mobile is-three-quarters-tablet"
          >
            <p
              className="user-page-card-name"
            >
              Awesome Northcoder
            </p>
            <p
              className="user-page-card-username"
            >
              @northcoder
            </p>
          </div>
        </div>
      </div>
      <div
        className="card-content content-container"
      >
        <div
          className="columns is-mobile"
        >
          <div
            className="column is-2-tablet"
          >
            <div
              className="user-page-card-mini-title"
            >
              Popularity
            </div>
            <div
              className="user-page-card-number"
            >
              NaN
            </div>
          </div>
          <div
            className="column is-2-tablet"
          >
            <div
              className="user-page-card-mini-title"
            >
              Comments
            </div>
            <div
              className="user-page-card-number"
            />
          </div>
          <div
            className="column is-2-tablet"
          >
            <div
              className="user-page-card-mini-title"
            >
              Articles
            </div>
            <div
              className="user-page-card-number"
            >
              5
            </div>
          </div>
          <div
            className="column is-2-tablet is-hidden-mobile"
          >
            <div
              className="user-page-card-mini-title"
            >
              Comment Votes
            </div>
            <div
              className="user-page-card-number"
            />
          </div>
          <div
            className="column is-2-tablet is-hidden-mobile"
          >
            <div
              className="user-page-card-mini-title"
            >
              Article Votes
            </div>
            <div
              className="user-page-card-number"
            />
          </div>
          <div
            className="column is-2-tablet is-hidden-mobile"
          >
            <div
              className="user-page-card-mini-title"
            >
              Ranking
            </div>
            <div
              className="user-page-card-number"
            >
              2
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
